From 85b018dc4c272444db5ad32c314afe3055d93183 Mon Sep 17 00:00:00 2001
From: Khiem Nguyen <khiem.nguyen.xt@renesas.com>
Date: Wed, 4 Oct 2023 19:21:40 +0300
Subject: [PATCH 19/40] drm: rcar-rvgc: Remove drm_fbdev_cma functions

The drm_fbdev_cma functions become obsoleted and removed since v5.0.
commit 2de304b4 (drm/cma-helper: Remove unused fbdev code).
Hence, following the changes in rcar-du changes at commit 2f690fad23
(drm/rcar-du: Use drm_fbdev_generic_setup()) to this driver.

Signed-off-by: Khiem Nguyen <khiem.nguyen.xt@renesas.com>
Signed-off-by: Vladyslav Andrishko <vladyslav.andrishko@globallogic.com>
---
 drivers/gpu/drm/rcar-rvgc/rcar_rvgc_drv.c  | 13 +++----------
 drivers/gpu/drm/rcar-rvgc/rcar_rvgc_drv.h  |  1 -
 drivers/gpu/drm/rcar-rvgc/rcar_rvgc_kms.c  | 13 -------------
 drivers/gpu/drm/rcar-rvgc/rcar_rvgc_pipe.c |  1 -
 4 files changed, 3 insertions(+), 25 deletions(-)

diff --git a/drivers/gpu/drm/rcar-rvgc/rcar_rvgc_drv.c b/drivers/gpu/drm/rcar-rvgc/rcar_rvgc_drv.c
index a82be701de7c..1bdbd256a19b 100644
--- a/drivers/gpu/drm/rcar-rvgc/rcar_rvgc_drv.c
+++ b/drivers/gpu/drm/rcar-rvgc/rcar_rvgc_drv.c
@@ -21,6 +21,7 @@
 #include <drm/drm_crtc_helper.h>
 #include <drm/drm_fb_cma_helper.h>
 #include <drm/drm_gem_cma_helper.h>
+#include <drm/drm_fb_helper.h>
 
 #include "rcar_rvgc_drv.h"
 #include "rcar_rvgc_kms.h"
@@ -107,12 +108,6 @@ static int rcar_rvgc_cb(struct rpmsg_device* rpdev, void* data, int len,
  * DRM operations
  */
 
-static void rcar_rvgc_lastclose(struct drm_device* dev) {
-	struct rcar_rvgc_device* rcrvgc = dev->dev_private;
-
-	drm_fbdev_cma_restore_mode(rcrvgc->fbdev);
-}
-
 static int rcar_rvgc_dumb_create(struct drm_file* file, struct drm_device* dev,
 				 struct drm_mode_create_dumb* args) {
 	unsigned int min_pitch = DIV_ROUND_UP(args->width * args->bpp, 8);
@@ -148,7 +143,6 @@ static struct drm_driver rcar_rvgc_driver = {
 	.desc			= "Renesas Virtual Graphics Card",
 	.date			= "20190408",
 	.fops			= &rcar_rvgc_fops,
-	.lastclose		= rcar_rvgc_lastclose,
 };
 
 
@@ -165,9 +159,6 @@ static void rcar_rvgc_remove(struct rpmsg_device* rpdev) {
 	else
 		kthread_stop(rcrvgc->vsync_thread);
 
-	if (rcrvgc->fbdev)
-		drm_fbdev_cma_fini(rcrvgc->fbdev);
-
 	if (rcrvgc->ddev) {
 		drm_dev_unregister(ddev);
 		drm_kms_helper_poll_fini(ddev);
@@ -257,6 +248,8 @@ static int rcar_rvgc_probe(struct rpmsg_device* rpdev) {
 	if (ret)
 		goto error;
 
+	drm_fbdev_generic_setup(ddev, 32);
+
 	DRM_INFO("Device %s probed\n", dev_name(&rpdev->dev));
 
 	return 0;
diff --git a/drivers/gpu/drm/rcar-rvgc/rcar_rvgc_drv.h b/drivers/gpu/drm/rcar-rvgc/rcar_rvgc_drv.h
index b50a7b3db281..e2b94882524a 100644
--- a/drivers/gpu/drm/rcar-rvgc/rcar_rvgc_drv.h
+++ b/drivers/gpu/drm/rcar-rvgc/rcar_rvgc_drv.h
@@ -32,7 +32,6 @@ struct rcar_rvgc_device {
 	struct device* dev;
 
 	struct drm_device* ddev;
-	struct drm_fbdev_cma* fbdev;
 
 	struct rpmsg_device* rpdev;
 
diff --git a/drivers/gpu/drm/rcar-rvgc/rcar_rvgc_kms.c b/drivers/gpu/drm/rcar-rvgc/rcar_rvgc_kms.c
index 9e09aeb8f36e..72399ae0788e 100644
--- a/drivers/gpu/drm/rcar-rvgc/rcar_rvgc_kms.c
+++ b/drivers/gpu/drm/rcar-rvgc/rcar_rvgc_kms.c
@@ -1,4 +1,3 @@
-#include <drm/drmP.h>
 #include <drm/drm_device.h>
 #include <drm/drm_atomic.h>
 #include <drm/drm_atomic_helper.h>
@@ -139,7 +138,6 @@ static const struct drm_mode_config_helper_funcs rcar_rvgc_mode_config_helper =
 
 int rcar_rvgc_modeset_init(struct rcar_rvgc_device* rcrvgc) {
 	struct drm_device* dev = rcrvgc->ddev;
-	struct drm_fbdev_cma* fbdev;
 	struct device_node* displays_node;
 	struct device_node* display_node;
 	int ret = 0;
@@ -274,17 +272,6 @@ int rcar_rvgc_modeset_init(struct rcar_rvgc_device* rcrvgc) {
 		}
 	}
 
-	/*
-	 * Initializes drm_fbdev_cma struct
-	 */
-	fbdev = drm_fbdev_cma_init(dev, 32, dev->mode_config.num_connector);
-	ret = IS_ERR(fbdev);
-	if (ret) {
-		dev_err(rcrvgc->dev, "drm_fbdev_cma_init failed: %d\n", ret);
-		goto exit;
-	}
-
-	rcrvgc->fbdev = fbdev;
  exit:
 	return ret;
 }
diff --git a/drivers/gpu/drm/rcar-rvgc/rcar_rvgc_pipe.c b/drivers/gpu/drm/rcar-rvgc/rcar_rvgc_pipe.c
index 9723a1b65388..17af33642f16 100644
--- a/drivers/gpu/drm/rcar-rvgc/rcar_rvgc_pipe.c
+++ b/drivers/gpu/drm/rcar-rvgc/rcar_rvgc_pipe.c
@@ -1,4 +1,3 @@
-#include <drm/drmP.h>
 #include <drm/drm_device.h>
 
 #include "rcar_rvgc_drv.h"
-- 
2.25.1

